Saule/Serialization/ResourceSerializer.cs

```diff
@@ -288,14 +292,16 @@ private JObject SerializeNode(ResourceGraphNode node, bool isCollection)
 
         private JObject SerializeAttributes(ResourceGraphNode node)
         {
+            var serializedSourceObject = JObject.FromObject(node.SourceObject, _sourceSerializer);
```

Comment:
This is not something I think we can do, because now we serialize *everything* on the object, rather than just what is required for the JSON API response. That could potentially include properties that should not be serialized because they represent a big graph or a relationship.

I think one of the main reasons of switching to the `ResourceGraph` concept was to avoid that type of bug.

Reply:
So inside the `SourceContractResolver` we do make sure to filter out the properties not assigned to the resource. Upon setting up the `sourceSerializer` we pass in the api resource to use.

Reply:
Ah, I see it now. Do you think we can add a comment on this line and line 315 to say that only the relevant attributes as defined in the `ApiResource` class are serialized?

Reply:
Yes of course, Ill also work on test for this all as well

Reply:
Great, thank you so much for working on this!

Reply:
Updated, np and thanks!